internal/provider/data_source_http_test.go

```diff
@@ -543,6 +543,29 @@ func CheckServerAndProxyRequestCount(proxyRequestCount, serverRequestCount *int)
 	}
 }
 
+func TestDataSource_Retry(t *testing.T) {
+	testHttpMock := setUpMockHttpServer(false)
+	defer testHttpMock.server.Close()
+
+	resource.UnitTest(t, resource.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+							data "http" "http_test" {
+								url = "%s/200"
+
+								retry_attempts = 1
+								retry_delay = 0
+							}`, testHttpMock.server.URL),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.http.http_test", "status_code", "200"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
Refer to [TestDataSource_Retry](https://github.com/hashicorp/terraform-provider-http/pull/151/files#diff-b661657abbfff7349d10226eb3a4f39991a71727446f826e4695147f73babb81R642) and [TestDataSource_MinDelay](https://github.com/hashicorp/terraform-provider-http/pull/151/files#diff-b661657abbfff7349d10226eb3a4f39991a71727446f826e4695147f73babb81R670) for additional test coverage for retries and delay between retries.